routes/api/events.js

```diff
@@ -1,9 +1,376 @@
+// @desc    Get all events
+// @access  Public
+router.get('/', async (req, res) => {
+  try {
+    const events = await Event.find();
+    res.json(events);
+  } catch (err) {
+    console.error(err.message);
+    res.status(500).send('Server Error');
+  }
+});
+
+// @route   GET api/events/m
+// @desc    Get all user's events
+// @access  Private
+router.get('/m', auth, async (req, res) => {
+  try {
+    const events = await Event.find({ user: req.user.id });
+    res.json(events);
+  } catch (err) {
+    console.error(err.message);
+    res.status(500).send('Server Error');
+  }
+});
+
+// @route   GET api/events/:id
+// @desc    Get event by ID
+// @access  Public
+router.get('/id/:id', async (req, res) => {
+  try {
+    const event = await Event.findById(req.params.id);
+
+    if (!event) {
+      return res.status(404).json({ msg: 'Event not found' });
+    }
+
+    res.json(event);
+  } catch (err) {
+    console.error(err.message);
+
+    if (err.kind === 'ObjectId') {
+      return res.status(404).json({ msg: 'Event not found' });
+    }
+
+    res.status(500).send('Server Error');
+  }
+});
+
+// @route   GET api/events/:id
+// @desc    Get all event participants
+// @access  Private
+router.get('/id/:id/participants', auth, async (req, res) => {
+  try {
+    const participants = await Event.find(
+      {
+        _id: req.params.id,
+        user: req.user.id,
+      },
+      {
+        participants: 1,
+      }
+    );
+
+    res.json(participants);
+  } catch (err) {
+    console.error(err.message);
+
+    if (err.kind === 'ObjectId') {
+      return res.status(404).json({ msg: 'Event not found' });
+    }
+
+    res.status(500).send('Server Error');
+  }
+});
+
+// @route   POST api/events
+// @desc    Create event
+// @access  Private
+router.post(
+  '/',
+  [
+    auth,
+    [
+      check('title', 'Title is required').notEmpty(),
+      check('description', 'Description is required').notEmpty(),
+      check('dateStart', 'Start date is required').notEmpty(),
+      check('dateEnd', 'End date is required').notEmpty(),
+      check('type', 'Type is required').notEmpty(),
+      check('location', 'Location is required').notEmpty(),
+    ],
+  ],
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    const {
+      title,
+      description,
+      dateStart,
+      dateEnd,
+      type,
+      location,
+      website,
+      image,
+      date,
+    } = req.body;
+
+    // Build event object
+    const eventFields = {};
+    eventFields.user = req.user.id;
+    if (title) eventFields.title = title;
+    if (description) eventFields.description = description;
+    if (dateStart) eventFields.dateStart = dateStart;
+    if (dateEnd) eventFields.dateEnd = dateEnd;
+    if (type) eventFields.type = type;
+    if (location) eventFields.location = location;
+    if (website) eventFields.website = website;

```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/spumony/xvent-project-backend/pullRequest?prid=7531905)
```suggestion
    if (website) {eventFields.website = website;}
```